sklearn/ensemble/weight_boosting.py

```diff
@@ -581,7 +581,7 @@ def _boost_discrete(self, iboost, X, y, sample_weight, random_state):
         # Only boost the weights if I will fit again
         if not iboost == self.n_estimators - 1:
             # Only boost positive weights
-            sample_weight = np.log(np.exp(sample_weight) +
+            sample_weight = np.exp(np.log(sample_weight) *
```

Comment:
no, you need a sum of logs. That * should be a +.

Reply:
oh! I am really sorry.